tensorflow/compiler/mlir/tensorflow/translate/mlir_roundtrip_flags.cc

```diff
@@ -190,11 +244,54 @@ Status ParseNodeNames(absl::string_view names_str,
   return Status::OK();
 }
 
+StatusOr<std::vector<std::string>> ParseDTypesHelper(
+    absl::string_view data_types_str) {
+  bool inside_subtype = false;
+  int cur_pos = 0;
+  std::vector<std::string> dtypes;
+  for (auto it : llvm::enumerate(data_types_str)) {
+    char c = it.value();
+    int i = it.index();
+    // Skip parsing the subtypes of a type
+    if (c == '(') {
+      if (inside_subtype) {
+        return errors::FailedPrecondition(
+            absl::StrCat("Syntax error in data types '", data_types_str, "'"));
```

Comment:
Can you add more descriptive error messages? At least something like "Unexpected '('".